### PR TITLE
Fix for `train.controlnet.controlnet_v1_5_1node_100steps`

### PR DESCRIPTION
**Pull request by rohitrango**
# What does this PR do ?

Fixes `train.controlnet.controlnet_v1_5_1node_100steps` CI test.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py [hunk 1]
```diff
@@ -1242,13 +1245,10 @@ def _load_pretrained_model(self, state_dict, ignore_mismatched_sizes=False, from
...
-                    s = key_.split('.')
-                    idx = int(s[-2])
-                    new_key_ = ".".join(s[:-2] + [str(int(idx + 1))] + [s[-1]])
-                    state_dict[key_] = state_dict[new_key_]
-                except:
-                    continue
+                s = key_.split('.')
+                idx = int(s[-2])
+                new_key_ = ".".join(s[:-2] + [str(int(idx + 1))] + [s[-1]])
+                state_dict[key_] = state_dict[new_key_]
```
Comment (Victor49152) [inline nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py:1252]: It's not a great practice but we do need this guard because in multiple usecases, this string s[-2] won't be a number to convert, it avoids throwing up an error.

Comment (Victor49152) [reply nemo/collections/multimodal/modules/stable_diffusion/diffusionmodules/openaimodel.py:1252]: So unless there is better solution, we need to keep this try statement.

**Review by Victor49152 — APPROVED** [review]

**Review by Victor49152 — APPROVED** [review]
